src/hooks/useAddressHistory.ts

```diff
@@ -64,6 +64,12 @@ export function useAddressHistory() {
   };
 
   useEffect(() => {
+    if (
+      (!isAddressEmpty && !id) ||
+      (!id && window.location.pathname === ROUTES.jetton + "/") ||
+      window.location.pathname === ROUTES.jetton
```

Comment:
I think that the route resolving should come from react router and not `window.location`